include/fiction/algorithms/simulation/sidb/quickexact.hpp

```diff
@@ -0,0 +1,506 @@
+                // dependent cell to change its charge state based on the N-1 SiDBs to fulfill the local population
+                // stability at its position.
+                charge_lyt_with_assigned_dependent_cell.update_after_charge_change(dependent_cell_mode::VARIABLE);
+
+                // If no positively charged SiDB can occur in the layout.
+                if (!three_state_simulation_required)
+                {
+                    result.additional_simulation_parameters.emplace_back("base_number", uint64_t{2});
+                    two_state_simulation(charge_lyt_with_assigned_dependent_cell);
+                }
+                // If positively charged SiDBs can occur in the layout, 3-state simulation is conducted.
+                else
+                {
+                    result.additional_simulation_parameters.emplace_back("base_number", uint64_t{3});
+                    three_state_simulation(charge_lyt_with_assigned_dependent_cell);
+                }
+            }
+            // if there is only one SiDB in the layout, this single SiDB can be neutrally or even positively charged due
+            // to external potentials or defects
+            else if (number_of_sidbs == 1)
+            {
+                if (three_state_simulation_required)
+                {
+                    charge_lyt.assign_base_number(3);
+                }
+                else
+                {
+                    charge_lyt.assign_base_number(2);
+                }
+
+                // A check is performed to see if the charge index is still below the maximum charge index. If not, the
+                // charge index is increased and the corresponding charge distribution is checked for physical validity.
+                while (charge_lyt.get_charge_index().first < charge_lyt.get_max_charge_index())
+                {
+                    if (charge_lyt.is_physically_valid())
+                    {
+                        charge_distribution_surface<Lyt> charge_lyt_copy{charge_lyt};
+                        result.charge_distributions.push_back(charge_lyt_copy);
+                    }
+
+                    charge_lyt.increase_charge_index_by_one(
+                        dependent_cell_mode::VARIABLE);  // "false" allows that the charge state of the dependent cell
+                                                         // is automatically changed based on the new charge
+                                                         // distribution.
+                }
+
+                if (charge_lyt.is_physically_valid())
+                {
+                    charge_distribution_surface<Lyt> charge_lyt_copy{charge_lyt};
+                    result.charge_distributions.push_back(charge_lyt_copy);
+                }
+            }
+            // If the layout consists of only detected negatively-charged SiDBs.
+            else if (all_sidbs_in_lyt_without_negative_detected_ones.empty() && number_of_sidbs > 1)
```

Comment:
Since you're repeating the `number_of_sidbs > 1` condition, couldn't you make that into one overarching condition, under which you test all the other parts individually?